src/bootstrap/builder.rs

```diff
@@ -1763,12 +1763,20 @@ impl<'a> Builder<'a> {
         );
 
         if !target.contains("windows") {
-            let needs_unstable_opts = target.contains("linux")
-                || target.contains("solaris")
-                || target.contains("windows")
-                || target.contains("bsd")
-                || target.contains("dragonfly")
-                || target.contains("illumos");
+            // FIXME(davidtwco): #[cfg(non(bootstrap))] - remove the `else` branch
+            let needs_unstable_opts = if compiler.stage >= 1 {
+                (target.contains("msvc")
+                    && self.config.rust_split_debuginfo != SplitDebuginfo::Packed)
+                    || (target.contains("windows")
+                        && self.config.rust_split_debuginfo != SplitDebuginfo::Off)
```

Comment:
Can this ever be true? We assert that `!target.contains("windows")` in the enclosing `if`, so neither `target.contains("msvc")` nor `target.contains("windows")` can be true, right?

Reply:
I've changed this logic in the most recent changes.